### PR TITLE
Add payload parser to darwin framework tool

### LINKED ISSUE #19676: [TC-DD-3.17] - DUT should terminate the commissioning process when we pass invalid QR Code, but DUT successfully completed the commissioning process
**Issue** : According to test-plan of [TC-DD-3.17] In Step 6b its mentioned DUT should terminate the commissioning process when we pass invalid 21 digit manual code generated in step 6a but DUT successfully completed the commissioning process.

**Expected behaviour** : DUT should terminate the commissioning mode when invalid 21 digit manual code is passed

**Actual behaviour** : DUT successfully completed the commissioning process.

**Commit used** : c1e1d018820c0fdd51f124f607b9207f31fae2ff			

**Steps to reproduce** :
**DUT side** :
sudo rm -rf /tmp/chip_*
sudo ./chip-all-clusters-app --wifi (Discriminator - 3841, Passcode - 20202021)

**TH side**
1. ./out/chip-tool/chip-tool pairing manualcode 1 749701123365521327694(0xFFF1)
2. ./out/chip-tool/chip-tool pairing manualcode 1 749701123365522327692(0xFFF2)
3.  ./out/chip-tool/chip-tool pairing manualcode 1 749701123365523327697(0xFFF3)
4. ./out/chip-tool/chip-tool pairing manualcode 1 749701123365524327693(0xFFF4)

**App used** - all-clusters-app

**Test platform** :
Chip-tool - RPI-4, 8GB RAM
DUT - RPI - RPI-4, 8GB RAM

**Test-plan reference** :
https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/devicediscovery.adoc#tc-dd-3-17-commissioning-flow-21-digit-manual-pairing-code-negative-scenario-dut-commissioner

**PFA Logs:**

[TC_DD_3.17_dut.txt](https://github.com/project-chip/connectedhomeip/files/8918788/TC_DD_3.17_dut.txt)
[TC_DD_3.17_cntrl.txt](https://github.com/project-chip/connectedhomeip/files/8918789/TC_DD_3.17_cntrl.txt)

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
Darwin-framework-tool does not have a payload parser to demonstrate that a commissioner implemented using CHIP.framework can indeed parse out the Vendor ID and check to see if it is valid. 
* Fixes #19676

#### Change overview
- Add a parser to Darwin-framework-tool

#### Testing
- Parsed SetupQRCode
- Parsed ManualCode

### CODE REVIEW

File: examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    }
+}
+
+CHIP_ERROR SetupPayloadParseCommand::Run()
+{
+    NSString * codeString = [NSString stringWithCString:mCode encoding:NSASCIIStringEncoding];
+    NSError * error;
+    CHIPSetupPayload * payload;
+    CHIPOnboardingPayloadType codeType;
+    if (IsQRCode(codeString)) {
```
Comment (bzbarsky-apple) [inline examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm:65]: As a followup: `CHIPOnboardingPayloadParser` should just do that check itself....

File: examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        NSLog(@"SerialNumber: %@", payload.serialNumber);
+    }
+    NSError * error;
+    NSArray<CHIPOptionalQRCodeInfo *> * optionalVendorData = [payload getAllOptionalVendorData:&error];
+    if (error) {
+        LogNSError("Error: ", error);
+        return CHIP_ERROR_INTERNAL;
+    }
+    for (const CHIPOptionalQRCodeInfo * info : optionalVendorData) {
+        bool isTypeString = info.infoType == [[NSNumber alloc] initWithInt:kOptionalQRCodeInfoTypeString];
```
Comment (bzbarsky-apple) [inline examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm:128]:
```suggestion
        bool isTypeString = info.infoType == @(kOptionalQRCodeInfoTypeString);
```
perhaps?  Same for the next line.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm (hunk 1 above)
Comment (krypton36) [reply examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm:65]: Ok. Created a new [issue](https://github.com/project-chip/connectedhomeip/issues/20311).

File: examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm (hunk 2 above)
Comment (krypton36) [reply examples/darwin-framework-tool/commands/payload/SetupPayloadParseCommand.mm]: Updated

**Pull request merged**